### PR TITLE
Conv1d subsampling modules: striding_conv1d and dw_striding_conv1d

### PR DESCRIPTION
**Pull request by burchim**
# What does this PR do ?

Add conv1d subsampling for ASR / lip reading

**Collection**: asr

# Changelog 
- Add 'striding_conv1d' and 'dw_striding_conv1d' subsampling options. Subsampling is performed using strided conv1d convolutions with kernel size 5 instead of doing reshape + frequency-temporal conv2d + reshape + linear projection.

# Usage
* Simply use 'subsampling: striding_conv1d in conformer yaml config for example.

```
  encoder:
      _target_: nemo.collections.asr.modules.ConformerEncoder
      feat_in: ${model.video_front_end.dim_output}
      feat_out: -1 # you may set it if you need different output size other than the default d_model
      n_layers: 18
      d_model: 512
  
      # Sub-sampling params
      subsampling: striding_conv1d # vggnet, striding, stacking or stacking_norm, dw_striding, striding_conv1d_k5, dw_striding_conv1d
      subsampling_factor: 8 # must be power of 2 for striding and vggne
...
```

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

File: nemo/collections/asr/parts/submodules/subsampling.py [hunk 1]
```diff
@@ -251,19 +251,128 @@ def __init__(
                     )
                 layers.append(activation)
                 in_channels = conv_channels
+
+        elif subsampling == 'striding_conv1d_k5':
```
Comment (titu1994) [inline nemo/collections/asr/parts/submodules/subsampling.py:255]: I'm not sure about the _k5 in the name. Best to remove it and keep it just ..._conv1d.

**Review by titu1994 — COMMENTED** [review]
Minor comment about name, but looks ok to me. @nithinraok for final review

File: nemo/collections/asr/parts/submodules/subsampling.py (hunk 1 above)
Comment (nithinraok) [reply nemo/collections/asr/parts/submodules/subsampling.py:255]: Agree with the same, pls remove _k5.

Comment (burchim) [reply nemo/collections/asr/parts/submodules/subsampling.py]: done, I removed the _k5 in new commit

**Comment by nithinraok** [comment]
Have you done any experiments with this subsampling method? I have done with FC, results are bit worse compared to 2D. Curious to see on other tasks.

**Review by nithinraok — APPROVED** [review]
LGTM

File: nemo/collections/asr/parts/submodules/subsampling.py (hunk 1 above)
Comment (nithinraok) [reply nemo/collections/asr/parts/submodules/subsampling.py]: thanks, update the change log accordingly.

**Comment by burchim** [comment]
@nithinraok I used this subsampling for lip reading experiments, since there are no reasons to use conv2d. I tried it a long time ago on Conformer but performance was a bit worse compared to using the classic conv2d

**Pull request closed without merging**